crates/deltalake-core/src/kernel/actions/types.rs

```diff
@@ -76,18 +76,16 @@ pub struct Metadata {
 impl Metadata {
     /// Create a new metadata action
     pub fn new(
-        id: impl Into<String>,
-        format: Format,
-        schema_string: impl Into<String>,
+        schema: StructType,
         partition_columns: impl IntoIterator<Item = impl Into<String>>,
-        configuration: Option<HashMap<String, Option<String>>>,
+        configuration: HashMap<String, Option<String>>,
     ) -> Self {
         Self {
-            id: id.into(),
-            format,
-            schema_string: schema_string.into(),
+            id: uuid::Uuid::new_v4().to_string(),
+            format: Default::default(),
+            schema_string: serde_json::to_string(&schema).unwrap(),
```

Comment:
It gives me the willies to see an unwrap in the constructor of `Metadata`, but since  this is accepting a `StructType`, is there any possible way for this to panic?

Reply:
rightfully so :) - this is removed locally now, and will be fixed in the next push...